src/components/Dashboard/OnType.jsx

```diff
@@ -1,29 +1,85 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import BigTile from './Tiles/BigTile';
 import Title from '../Typography/Title';
 import Clock from '../Clock';
+import Fuse from 'fuse.js';
+import { useAuth } from '../../contexts/AuthContextProvider';
+import { collection, query, getDocs } from 'firebase/firestore';
+import { db } from '../../utils/FirebaseConfig';
+
+import websiteList from '../../data/WebsiteList.json';
+
+const options = {
+	isCaseSensitive: false,
+	findAllMatches: true,
+	shouldSort: true,
+	minMatchCharLength: 1,
+	threshold: 0.49,
+	keys: ['hostname', 'title'],
+};
+
+const OnType = ({ params }) => {
+	const { user } = useAuth();
+	const [suggestions, setSuggestions] = useState([]);
+	const [userHistory, setUserHistory] = useState([]);
+
+	const fuseSearch = (list, input) => {
+		const fuse = new Fuse(list, options);
+		const result = fuse.search(input);
+		setSuggestions(result);
+	};
+
+	useEffect(() => {
+		setUserHistory(websiteList);
+		if (!user) return;
+
+		
+		const fetchData = async () => {
+			const q = query(collection(db, `users/${user.uid}`, 'history'));
+			const querySnapshot = await getDocs(q);
+
+			let history = [];
+			querySnapshot.forEach((doc) => {
+				// Check if the hostname is already in the list
+				const isInList = history.some((item) => item.hostname === doc.data().hostname);
+				if (!isInList) history.push(doc.data());
+			});
+
+			let websiteListFiltered = [];
+			websiteList.forEach((obj) => {
+				const isInList = history.some((item) => item.hostname === obj.hostname);
+				if (!isInList) websiteListFiltered.push(obj);
+			});
+
+			history = history.concat(websiteListFiltered);
+			setUserHistory(history);
+		};
+
+		fetchData();
+	}, [user]);
+
+	useEffect(() => {
+		const searchInput = params.get('search');
+		fuseSearch(userHistory, searchInput);
+	}, [params, userHistory]);
```

Comment:
fuseSearch moet mee in de dependency array. Maar je gaat fuseSearch dan in een useCallback moeten steken. Ofwel de fuseSearch functie buiten de component zetten en de results laten returnen